components/match2/wikis/mobilelegends/match_legacy.lua

```diff
@@ -0,0 +1,172 @@
+---
+-- @Liquipedia
+-- wiki=mobilelegends
+-- page=Module:Match/Legacy
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local MatchLegacy = {}
+
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+function MatchLegacy.storeMatch(match2)
+	local match = MatchLegacy._convertParameters(match2)
+
+	match.games = MatchLegacy.storeGames(match, match2)
+
+	MatchLegacy.storeMatchSMW(match, match2)
+
+	return mw.ext.LiquipediaDB.lpdb_match(
+		'legacymatch_' .. match2.match2id,
+		match
+	)
+end
+
+function MatchLegacy._convertParameters(match2)
+	local match = Table.deepCopy(match2)
+	for key, _ in pairs(match) do
+		if String.startsWith(key, 'match2') then
+			match[key] = nil
+		end
+	end
+	match.links = nil
+
+	if String.isNotEmpty(match.walkover) then
+		match.resulttype = match.walkover
+		match.walkover = match.winner
+	end
+
+	match.staticid = 'Legacy_' .. match2.match2id
+
+	-- Handle extradata fields
+	match.extradata = {}
+	local extradata = Json.parseIfString(match2.extradata)
+	match.extradata.matchsection = extradata.matchsection
+	match.extradata.mvpteam = extradata.mvpteam
+	match.extradata.mvp = extradata.mvp
+	match.extradata.comment = extradata.comment
+
+	local opponents = match2.match2opponents or {}
+	match.extradata.team1icon = (opponents[1] or {}).icon
+	match.extradata.team2icon = (opponents[2] or {}).icon
+	match.extradata.opponent1literal = tostring((opponents[1] or {}).type == 'literal')
+	match.extradata.opponent2literal = tostring((opponents[2] or {}).type == 'literal')
+	match.extradata.opponent1game = 'false'
+	match.extradata.opponent2game = 'false'
+
+	local games = match2.match2games or {}
+	for key, game in ipairs(games) do
+		if String.isNotEmpty(game.vod) then
+			match.extradata['vodgame' .. key] = game.vod
+		end
+	end
+
+	match.extradata = mw.ext.LiquipediaDB.lpdb_create_json(match.extradata)
+
+	-- Handle Opponents
+	local handleOpponent = function (index)
+		local prefix = 'opponent'..index
+		local opponent = match2.match2opponents[index] or {}
+		local opponentmatch2players = opponent.match2players or {}
+		if opponent.type == 'team' then
+			match[prefix] = opponent.name
+			match[prefix.."score"] = (tonumber(opponent.score) or 0) > 0 and opponent.score or 0
```

Comment:
change the `"` to `'` to match the style in the rest of this module. 
Same on the other place with score.